app/src/script.js

```diff
@@ -105,39 +118,56 @@ async function updateConnectedAccount(state, { account }) {
 
 async function newTokenRequest(
   state,
-  { requestId, requesterAddress, depositToken, depositAmount, requestAmount, date },
+  { requestId, requesterAddress, depositToken, depositAmount, requestAmount },
   settings,
   blockNumber
 ) {
-  const { account, requests } = state
-  let status
-  if (!account) return state
+  try {
+    const { account, requests } = state
+    let status
+    if (!account) return state
+    let decimals
+    let name
+    let symbol
+    if (depositToken === ETHER_TOKEN_FAKE_ADDRESS) {
+      decimals = 18
+      name = 'Ether'
+      symbol = 'ETH'
+    } else {
+      const depositTokenData = await getTokenData(depositToken, settings)
+      decimals = depositTokenData.decimals
+      name = depositTokenData.name
+      symbol = depositTokenData.symbol
+    }
```

Comment:
We could declared a constant object at the top of the script with the Ether data and the change this to
```
const { decimals, name, symbol } = depositToken === ETHER_TOKEN_FAKE_ADDRESS ? ETHER_DATA : (await getTokenData(depositToken, settings))
```